TUSKit/Classes/TUSExecutor.swift

```diff
@@ -212,9 +214,8 @@ class TUSExecutor: NSObject, URLSessionDelegate {
         if continueUploading() {
             let pendingUploads = TUSClient.shared.pendingUploads()
             TUSClient.shared.createOrResume(forUpload: pendingUploads[0])
-        } else {
-            completion(true)
         }
+        completion(true)
```

Comment:
hmm, I am unsure about this change. Calling completion will stop the background task. So, even when we are calling `continueUploading` the background task will stop.
This code:
```
            let pendingUploads = TUSClient.shared.pendingUploads()
            TUSClient.shared.createOrResume(forUpload: pendingUploads[0])
```
is being run in the background task we just stopped 🤔 
Or am I missing something?

Reply:
That change is in `handleUploadSuccess(upload:completion:)` method which means that upload is completed and background task should be closed. 

In case there are more files to upload, `createOrResume(forUpload:)` method will be triggered, which opens new background task for that upload. This should work well in foreground state, but remains to be seen if this is possible when app goes to background and `TUSBackgroundMode.PreferUploadQueue` is set as `backgroundMode`. In my opinion `PreferUploadQueue` should be avoided in most situations as there are no guarantees that whole queue will be uploaded on time allocated by iOS.

As a proof for this issue, console warnings of still running background tasks can be observed in case when there is more than one file in upload queue. Number of background task warnings equals to `N-1`, where `N` is the number of files in queue. `N-1` is because only last file in queue closes its background task.

While we are at it, same situation is in markAsFailed(upload:completion:) method.